velox/dwio/common/DecoderUtil.h

```diff
@@ -163,7 +163,8 @@ void fixedWidthScan(
     const char*& bufferStart,
     const char*& bufferEnd,
     TFilter& filter,
-    THook& hook) {
+    THook& hook,
+    int32_t numValuesBeforePage) {
```

Comment:
You should pass in `scatterRows + numValuesBeforePage` in the call sites instead

Reply:
Only when the hook is set, `numValuesBeforePage` is required. Otherwise, it is not needed, so I pass it to the function for processing.

Reply:
Then there must be something wrong how `scatterRows` is set up.  It should be same offset for hook and read values

Reply:
I think your analysis is correct, because when I added the page offset to the hook logic, some tests passed, but some still failed.

Reply:
Can you pass in a `scatterRow` with all values shifted by `numValuesBeforePage`?  I don't think `fixedWidthScan` and `processFixedWidthRun` should be concerned about this offset

Reply:
Already refactored.